v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/transforms/SourceWriterFn.java

```diff
@@ -373,4 +338,37 @@ void outputWithTag(
     }
     c.output(tag, gson.toJson(errorRecord, ChangeStreamErrorRecord.class));
   }
+
+  TupleTag<String> classifyException(Exception exception) {
+    if (exception instanceof SpannerException e) {
+      return classifySpannerException(e);
+    } else if (exception instanceof ChangeEventConvertorException) {
+      return Constants.PERMANENT_ERROR_TAG;
+    }
+    return Constants.RETRYABLE_ERROR_TAG;
```

Comment:
please add an info log here. If the code is reaching the parts which is not handled explicitly, then there should be a way to trace that.

Reply:
Resolving the comment as of now since we need this fix urgently. We can come back and discuss about this later.